staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery_test.go

```diff
@@ -437,3 +442,48 @@ func fetchPath(handler http.Handler, etag string) (*http.Response, []byte, *apid
 
 	return w.Result(), bytes, decoded
 }
+
+// completerWorkqueue is a workqueue.RateLimitingInterface that implements
+// isComplete
+type completerWorkqueue struct {
+	lock sync.Mutex
+	workqueue.RateLimitingInterface
+	processing map[interface{}]struct{}
+}
+
+var _ = workqueue.RateLimitingInterface(&completerWorkqueue{})
+
+func newCompleterWorkqueue(wq workqueue.RateLimitingInterface) *completerWorkqueue {
+	return &completerWorkqueue{
+		RateLimitingInterface: wq,
+		processing:            make(map[interface{}]struct{}),
+	}
+}
+
+func (q *completerWorkqueue) Add(item interface{}) {
+	q.lock.Lock()
+	defer q.lock.Unlock()
+	q.processing[item] = struct{}{}
+	q.RateLimitingInterface.Add(item)
+}
+
+func (q *completerWorkqueue) AddAfter(item interface{}, duration time.Duration) {
+	q.Add(item)
+}
+
+func (q *completerWorkqueue) AddRateLimited(item interface{}) {
+	q.Add(item)
+}
```

Comment:
This feels weird to me, but I know neither of these ever are called by discovery manager, since [`syncAPIService`](https://github.com/kubernetes/kubernetes/blob/40c5b1b7c37804f5c8656dffadf7f086daa0a9ba/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery.go#L344) unconditionally returns nil. (Discovery Manager refreshes all services on minute interval unconditionally, so does not make use of rate limiter)

I'm wondering if this or a future PR should refactor it to not use the `workqueue.RateLimitingInterface` and instead use normal `workqueue.Interface`. I think we can keep that in mind for a separate PR, to retain this PR's nice property of only changing test files.